### PR TITLE
feat(buildkite): add buildkite charts

### PR DESCRIPTION
**Pull request by rimusz**

### CODE REVIEW

File: buildkite-agent-private-git/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: buildkite-agent-private-git
+home: https://github.com/buildkite/agent
+version: 1.0.0
+description: Buildkite Agent for private Git repositories
+maintainers:
+  - Tim Lucas <tim@buildkite.com>
```
Comment (arschles) [inline buildkite-agent-private-git/Chart.yaml]: @rimusz aren't you the maintainer of this chart?

File: buildkite-agent-private-git/manifests/buildkite-agent-rc.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          env:
+            - name: BUILDKITE_AGENT_TOKEN
+              value: "{{ .token }}"
+            - name: SSH_PRIVATE_RSA_KEY_PATH
+              value: "/mnt/ssh/id-rsa"
+          volumeMounts:
+            - name: ssh-secret
+              mountPath: /mnt/ssh
+            - name: var-run-docker-sock
+              mountPath: /var/run/docker.sock
```
Comment (arschles) [inline buildkite-agent-private-git/manifests/buildkite-agent-rc.yaml:29]: @rimusz can you modify the `README.md` to indicate that this chart assumes all hosts are configured with the Docker socket at `/var/run/docker.sock`?

File: buildkite-agent/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+name: buildkite-agent
+home: https://github.com/buildkite/agent
+version: 1.0.0
+description: Buildkite Agent
+maintainers:
+  - Tim Lucas <tim@buildkite.com>
```
Comment (arschles) [inline buildkite-agent/Chart.yaml]: @rimusz same comment here RE you being the maintainer

File: buildkite-agent/manifests/buildkite-agent-rc.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+          image: buildkite/agent
+          imagePullPolicy: Always
+          securityContext:
+            privileged: true
+          env:
+            - name: BUILDKITE_AGENT_TOKEN
+              value: "{{ .token }}"
+          volumeMounts:
+            - name: var-run-docker-sock
+              mountPath: /var/run/docker.sock
```
Comment (arschles) [inline buildkite-agent/manifests/buildkite-agent-rc.yaml:25]: @rimusz same comment here RE adding to the `README.md` to indicate the location of the Docker socket on hosts

File: buildkite-agent-private-git/manifests/buildkite-agent-rc.yaml (hunk 2 above)
Comment (rimusz) [inline buildkite-agent-private-git/manifests/buildkite-agent-rc.yaml:29]: sure will do both

File: buildkite-agent/manifests/buildkite-agent-rc.yaml (hunk 4 above)
Comment (rimusz) [inline buildkite-agent/manifests/buildkite-agent-rc.yaml:25]: readme.md for both charts got as per @arschles comment, forgot to mention it

```
Node hosts need to be configured to run with the Docker socket at /var/run/docker.sock
 for this chart to work properly.
```

Comment (rimusz) [inline buildkite-agent/manifests/buildkite-agent-rc.yaml:25]: @mboersma can you take a  look so we can finally release it

**Comment by rimusz** [comment]
@arschles @mboersma can I get LGTM for this one, as all requested changes are done

**Comment by rimusz** [comment]
Helm Classic deprecated, closing this

**Pull request closed without merging**